File: leetcode/src/MergeSortedArray.java
import java.util.ArrayList;
import java.util.List;

public class MergeSortedArray {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        if (n == 0) {

        } else if (m == 0) {
            for (int i = 0; i < nums2.length; i++) {
                nums1[i] = nums2[i];
            }
        } else {
            List<Integer> list = new ArrayList<Integer>();
            int nums1Index = 0;
            int nums2Index = 0;
            while (true) {
                if (nums2Index == n && nums1Index == m) {
                    break;
                }
                if (nums2Index == n) {
                    list.add(nums1[nums1Index++]);
                } else if (nums1Index == m) {
                    list.add(nums2[nums2Index++]);
                } else if (nums1[nums1Index] <= nums2[nums2Index]) {
                    list.add(nums1[nums1Index++]);
                } else if (nums1[nums1Index] >= nums2[nums2Index]) {
                    list.add(nums2[nums2Index++]);
                }
            }
            for (int i = 0; i < list.size(); i++) {
                nums1[i] = list.get(i);
            }
        }
    }
}

File: leetcode/src/python/merge_sorted_array.py
from typing import List

class MergeSortedArray:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        if n != 0:
            index1 = 0
            index2 = 0
            while index2 < n:
                if index1 == m + index2:
                    nums1[index1] = nums2[index2]
                    index2 += 1
                else:
                    num1 = nums1[index1]
                    num2 = nums2[index2]
                    if num2 <= num1:
                        nums1.insert(index1, num2)
                        nums1.pop(len(nums1) - 1)
                        index2 += 1
                    index1 += 1

if __name__ == '__main__':
    instance = MergeSortedArray()
    nums1 = [1, 2, 3, 0, 0, 0]
    m = 3
    nums2 = [2, 5, 6]
    n = 3
    # nums1 = [0]
    # m = 0
    # nums2 = [1]
    # n = 1
    instance.merge(nums1, m, nums2, n)
    print(nums1)